### PR TITLE
[TECHBRAIN-318] Implement build ejs to html file

### PR DESCRIPTION
**Pull request by gayatri195**
## PR to close at the same time

- URL

### List changes you made

1. Implemented build ejs to html file
2. Added build.js to compile ejs into html

### Explain why you made those changes

1. 
2. 

## Checklist

- [x] No conflict?
- [ ] Postman request added if new endpoint's implemented?
    - If no, reason: 
- [ ] Spec added for new feature?
    - If no, reason: 
- [ ] Spec passed?
    - If no, reason: 
- [ ] Lint your code?
    - If no, reason: 
- [ ] Tested in a shared environment?
    - If no, reason: 

## Environment to test

- [ ] local env
- [ ] testing
- [ ] staging

## Steps to reproduce

1. 
2. 

## Points to reviews

- things reviewers need to look into

### CODE REVIEW

**Comment by parashagrawal** [comment]
Add PR template

File: build.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const ejs = require('ejs');
+const fs = require('fs');
+const path = require('path');
+
+// Define the input and output directories
+const ejsDirectory = path.join(__dirname, 'views', 'ejs');
```
Comment (parash-namespace) [inline build.js:6]: This line is not used so we can remove it

File: build.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const htmlDirectory = path.join(__dirname, 'views', 'html');
```
Comment (parash-namespace) [inline build.js:7]: We can directly define `htmlFilePath` instead of `htmlDirectory`

File: build.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+// Function to compile the index.ejs file
+const compileIndexEjs = () => {
+  const indexEjsFilePath = path.join(__dirname, 'index.ejs');
```
Comment (parash-namespace) [inline build.js:11]: Move it out of the block

File: index.ejs [hunk 4]
```diff
@@ -2,50 +2,50 @@
 <html lang="en">
 
 <head>
-  <%- include('partials/meta.ejs') %>
+  <%- include('views/ejs/partials/meta.ejs') %>
 
-    <%- include('partials/links.ejs') %>
+    <%- include('views/ejs/partials/links.ejs') %>
```
Comment (parashagrawal) [inline index.ejs]: Indentation

File: build.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    fs.writeFileSync(htmlFilePath, html, 'utf-8');
+
+    console.log(`Compiled ${indexEjsFilePath} to ${htmlFilePath}`);
+  } else {
+    console.error('index.ejs file not found.');
+  }
+};
+
+compileIndexEjs();
```
Comment (parashagrawal) [inline build.js]:
```suggestion
const ejs = require('ejs');
const fs = require('fs');
const path = require('path');

const htmlFilePath = path.join(__dirname, 'views', 'html', 'index.html');
const indexEjsFilePath = path.join(__dirname, 'index.ejs');

// Function to compile the index.ejs file
const compileIndexEjs = () => {
  if (fs.existsSync(indexEjsFilePath)) {
    const ejsContent = fs.readFileSync(indexEjsFilePath, 'utf-8');
    fs.mkdirSync(path.dirname(htmlFilePath), { recursive: true });
    const html = ejs.render(ejsContent, { filename: indexEjsFilePath });
    fs.writeFileSync(htmlFilePath, html, 'utf-8');
    console.log(`Compiled ${indexEjsFilePath} to ${htmlFilePath}`);
  } else {
    console.error('index.ejs file not found.');
  }
};

compileIndexEjs();
```

**Review by parashagrawal — CHANGES_REQUESTED** [review]

**Review by parashagrawal — APPROVED** [review]

**Pull request merged**